index.js

```diff
@@ -502,6 +503,16 @@ function urlToOptions(urlObject) {
   return options;
 }
 
+function getMatchingHeader(regex, headers) {
+  var lastValue;
+  for (var header in headers) {
+    if (regex.test(header)) {
+      lastValue = headers[header];
+    }
+  }
+  return lastValue;
```

Comment:
This function should instead be implemented as a reverse loop, returning as soon as a match is found.

Reply:
no longer needed